notebooks/test.sh

```diff
@@ -24,14 +26,22 @@ for track in $TRACKS
 do
     # Run each step of the rendering pipeline, to make sure it runs without errors.
     python3 clean.py $track
+    # If this fails (i.e. if running clean.py results in changes to the working tree),
+    # this means that clean.py wasn't run before committing.
+    git diff --exit-code --no-color -- $track
     python3 prepare_push.py $track
-    # TODO: Fails due to failure when calling get_git_branch()
-    #python3 render.py $track
+    python3 render.py $track
 done
 
 TESTABLE_NOTEBOOK_TRACKS="deep_learning python pandas machine_learning"
 for track in $TESTABLE_NOTEBOOK_TRACKS
 do
+    # Running the deep learning notebooks is fairly slow (~10-20 minutes), so only
+    # do it if any of the relevant files have changed between this branch and master.
+    if [[ $track == "deep_learning" ]] && git diff --exit-code master -- $track ../learntools/$track; then
```

Comment:
Interested in your thoughts about why you restricted this to deep learning?

Tradeoff seems to be: 
- Running unchanged tracks allows us to catch failures introduced by updates in other packages (like pandas)

- Running unchanged courses adds time to our tests

Does that sound like an accurate description of the tradeoff?  I ask in part to inform what else I include in this line in the future, if we find we have other tracks that take a non-trivial amount of time.

Reply:
> Running unchanged tracks allows us to catch failures introduced by updates in other packages (like pandas)

Yeah, and also failures introduced by changes to learntools.core or the notebook pipeline. As long as running a track's notebooks is fairly cheap, I think it's worth doing every time.